process/transaction/shardProcess.go

```diff
@@ -718,7 +718,12 @@ func (txProc *txProcessor) executeFailedRelayedTransaction(
 		return err
 	}
 
-	totalFee := txProc.economicsFee.ComputeFee(userTx)
+	totalFee := core.SafeMul(userTx.GetGasPrice(), userTx.GetGasLimit())
```

Comment:
So, this will take all the gas set in userTx for in shard relayed txs? Could be a situation when GasLimit is set to 0 and we will not take anything, or worse in cross shard call if totalFee is lower than txProc.economicsFee.ComputeMoveBalanceFee(userTx), then in line 724 we will result in a negative value?

Reply:
No. there is no way to have that. This is the correct way, until now that computation did only movebalance transfer cost which was always smaller than equal to the actual one.

Reply:
ok